addons/token-exchange/agent_mirrorpeer_controller.go

```diff
@@ -84,13 +88,71 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, fmt.Errorf("failed to enable mirroring the storagecluster %q in namespace %q in managed cluster. Error %v", scr.Name, scr.Namespace, err)
 	}
 
+	err = r.createS3(ctx, req)
+	if err != nil {
+		klog.Error(err, "Failed to create ODR S3 resources")
+		return ctrl.Result{}, err
+	}
+
 	err = r.createVolumeReplicationClass(ctx, &mirrorPeer)
 	if err != nil {
 		return ctrl.Result{}, fmt.Errorf("failed to create volumereplicationclass: %v", err)
 	}
 	return ctrl.Result{}, nil
 }
 
+func (r *MirrorPeerReconciler) createS3(ctx context.Context, req ctrl.Request) error {
+	var err error
+
+	var mirrorPeer multiclusterv1alpha1.MirrorPeer
+	err = r.HubClient.Get(ctx, req.NamespacedName, &mirrorPeer)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			klog.Info("Could not find MirrorPeer")
+			return err
+		}
+		klog.Error(err, "Failed to get MirrorPeer")
+		return err
+	}
```

Comment:
MirrorPeer is already available in Reconcile function. Use that instead of fetching it again.